src/Components/Onboarding/Steps/Artists/ItemLink.tsx

```diff
@@ -62,16 +62,16 @@ class ItemLink extends React.Component<Props, State> {
   }
 
   followArtist(artist: ArtistFollowProps) {
-    const storeUpdater = (store: RecordSourceSelectorProxy) => {
-      const mutationPayload = store.getRootField("followArtist")
-      const origArtist = mutationPayload.getLinkedRecord("artistEdge")
-
-      // Useful to import this directly for debugging
-      const localConnHandler = ConnectionHandler
-
-      // Somehow update the artist with the suggestion...
-      const selectedArtist = store.get(artist.__id)
-      const connection = localConnHandler.getConnection(origArtist, "ItemLink_suggested")
+    const storeUpdater = (store: RecordSourceSelectorProxy, data: any) => {
+      // Search the store for a record for the newly suggested artist, based on its ID we get from the response `data`.
+      const suggestedArtist = store.get(data.followArtist.artistEdge.node.related.suggested.edges[0].node.__id)
+      // Replace `artist` with the `suggestedArtist` in the list of `artists` on the `popular_artists` root field.
+      const popularArtistsRootField = store.get("client:root:popular_artists")
```

Comment:
Btw, this currently hardcodes `popular_artists` but this component is used in both the popular artists _and_ search results list, so the identifier should probably be passed in by the parent using this component.